common/src/collision.rs

```diff
@@ -23,54 +23,122 @@ pub fn shape_cast(
     start_node_transform: na::Matrix4<f32>,
     ray: &Ray,
     tanh_distance: f32,
-) -> RayEndpoint {
+) -> Result<RayEndpoint, RayTracingError> {
     // A collision check is assumed to be a miss until a collision is found.
     // This `endpoint` variable gets updated over time before being returned.
     let mut endpoint = RayEndpoint {
         tanh_distance,
         hit: None,
     };
 
+    // Start a breadth-first search of the graph's chunks, performing collision checks in each relevant chunk.
+    // The `chunk_queue` contains ordered pairs containing the `ChunkId` and the transformation needed to switch
+    // from the original node coordinates to the current chunk's node coordinates.
+    let mut visited_chunks: HashSet<ChunkId> = HashSet::new();
```

Comment:
nit: `FxHashSet` is a better choice when the input isn't at risk of being carefully crafted to produce a DoS.

Reply:
I originally chose `HashSet` because this is a test. I'll switch it over.

Reply:
`pub fn shape_cast` is a test?

Reply:
Oh, whoops. I got this mixed up with the `VoxelAABB` test.